anyway/infographic_image_generator.py

```diff
@@ -34,53 +41,90 @@ def get_download_button_elements(browser):
     return svgs
 
 
-def wait_for_folder_to_contain_all_files(download_directory, number_of_expected_files, timeout):
+def get_local_infographics_folder_name(newsflash_id):
+    return f"./{INFOGRAPHICS_LOCAL_FOLDER}/{newsflash_id}"
+
+
+def get_unique_filenames(filenames):
+    names = set()
+    for filename in filenames:
+        index = filename.find(" ")
+        if index > -1:
+            base_filename = filename[:index]
+            extension = filename.split(".")[1]
+            filename = base_filename + "." + extension
+        names.add(filename)
+    return list(names)
+
+
+def wait_for_folder_to_contain_all_files(newsflash_id, number_of_expected_files, timeout):
     for _ in range(timeout):
         time.sleep(1)
-        files = os.listdir(download_directory)
-        if len(files) == number_of_expected_files:
-            return True
-    return False
+        contents = requests.get(
+            f"{selenium_remote_results_url}/{newsflash_id}/").json()
+        filenames = [item['name'] for item in contents]
+        unique_filenames = get_unique_filenames(filenames)
+        if len(unique_filenames) == number_of_expected_files:
```

Comment:
I think there is a remote possibility that this function will terminate after the last file was created, but before it was completely downloaded.

Reply:
yes, well, this function is supposed to try and wait until everything is downloaded, the question is what the timeout should be. usually 1 is enough, but we can make it let's say 60 seconds to be sure. what do you think?

Reply:
Ok, I see your point. On the discussion below they say that Chrome uses an extension for partial files that were not completely downloaded yet, so I added a check that no file has this extension.

https://stackoverflow.com/questions/34338897/python-selenium-find-out-when-a-download-has-completed

Reply:
@ziv17 @tkalir 
Great work.
I approved and merged. However if you think this issue should be further discussed, let's open a new issue.